api/src/lib/auth.js

```diff
@@ -6,8 +6,18 @@
 //   }
 
 import { AuthenticationError } from '@redwoodjs/api'
+import * as emailpassword from "supertokens-node/recipe/emailpassword";
 
 export const getCurrentUser = async (_decoded, { type, token }) => {
+  if (type === "supertokens") {
+    return {
+      type,
+      userId: _decoded.getUserId(),
+      jwtPayload: _decoded.getJWTPayload(),
+      sessionHandle: _decoded.getHandle(),
+      email: (await emailpassword.getUserById(_decoded.getUserId())).email
+    }
+  }
```

Comment:
For users of supertokens, the `_decoded` variable will be a `session` object (result of session verification). More info about this object can be found here: https://supertokens.io/docs/session/common-customizations/sessions/session-verification#reqsession-object